arrow/formatter.py

```diff
@@ -25,6 +26,11 @@ def format(cls, dt, fmt):
 
     def _format_token(self, dt, token):
 
+        if token and token.startswith("[") and token.endswith("]"):
+            return self._ESCAPED_BRACKET_RE.sub(
+                lambda m: m.group(1), token[1:-1]
```

Comment:
Could you explain exactly what the lambda function is doing here? Maybe adding a few comments to the source code would help.